debugger/src/debugger.rs

```diff
@@ -1435,10 +1493,20 @@ pub fn debug(debugger_options: DebuggerOptions, dap: bool, vscode: bool) {
                                 }
                                 Err(_) => {
                                     log::error!("Failed to negotiate non-blocking socket with request from :{}", addr);
+                                    return;
```

Comment:
Maybe it would make sense to return an error directly from this function? Then we could just use `?` to return errors, which would simplify this function a lot.

Reply:
This code is inside a loop, which starts and stops with a session, and then waits again for a the next session to connect. I evealuate the errors here, to make decisions about continuing the process or not. The `?` would exit the loop and the function - wouldn't this just move the code of evaluating the error to the caller?

Reply:
Something I didn't clarify before. Because the debugger is 'hidden' from the user (it is run in the background by VSCode), my design approach has been to make it as resilient to errors as possible. In other words, catch and understand all possible user errors, give the feedback, and allow the user to try again - exiting the process with an error gives a bit of annoying user experience IMHO :)

Reply:
> 
> 
> This code is inside a loop, which starts and stops with a session, and then waits again for a the next session to connect. I evealuate the errors here, to make decisions about continuing the process or not. The `?` would exit the loop and the function - wouldn't this just move the code of evaluating the error to the caller?

The `?` would exit the function, that's right. For errors where you don't want to exit the function, it cannot be used, but for the other cases, were a `return` is used, I think it would be a good fit. For TCP / communication errors, I'm not sure if you can give a lot of useful feedback to the user, since the only method of communication is broken. Just exiting is probably the best solution for that case.

Something I also like to do for theses cases is to create a new function, which would contain the current loop body. In that function, you can then return an error without problems, and in the loop you can then still decide to continue with the loop, or to exit completely.

I'm also fine with keeping the current implementation, I think a lot of this is just down to taste 😄 .

Reply:
Thank you @Tiwalun ... I will keep with the current solution for now, but appreciate your attentive review a lot. I learn something every time!